blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -143,7 +142,11 @@ public Link getLink() {
 
     @Override
     public String getFullName(){
-        return job.getFullName();
+        // Lets make sure that the fullName is properly encoded etc.
+        // The name part of the fullName can contain unencoded characters
+        // if the Job was programmatically created, causing screwups
+        // when mapping those URLs to Blue Ocean web/rest URLs. See JENKINS-41425.
+        return BlueUrlTokenizer.reconstructJobNamePath(job.getName(), job.getFullName());
```

Comment:
BluePipeline.getName() should be encoded as well. This is to keep things consistent between job name created using UI vs creating them programatically.